src/Sarif/PeekableStream.cs

```diff
@@ -0,0 +1,129 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif.Readers
+{
+    /// <summary>
+    /// This stream provides a very limited Seekability for Streams which would benefit from an ersatz Seek(Origin)
+    /// capability, but which only need it at the beginning of a Stream.  Ie: to categorize binary versus textual 
```

Comment:
>Ie [](http://example.com/codeflow?start=74&length=2)

s/be rendered as `i.e.` if you prefer this. Also this is a sentence fragment, maybe go ahead and append to the previous sentence after a comma.